File: app/src/main/java/com/chiranths/jobportal1/Activities/BasicActivitys/SeeAllLayoutActivity.java
package com.chiranths.jobportal1.Activities.BasicActivitys;

import android.os.AsyncTask;
import android.os.Build;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.chiranths.jobportal1.Activities.Propertys.PropertyActivity;
import com.chiranths.jobportal1.Adapters.AdsAdaptor;
import com.chiranths.jobportal1.Adapters.LayoutsAdaptor;
import com.chiranths.jobportal1.Adapters.SeeallLayouts;
import com.chiranths.jobportal1.R;
import com.chiranths.jobportal1.Utilitys.AppConstants;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;

public class SeeAllLayoutActivity extends AppCompatActivity implements View.OnClickListener {
    private RecyclerView recyclerView;
    RecyclerView.LayoutManager layoutManager;
    List layoutlist = new ArrayList();
    private Handler mHandler = new Handler();
    SeeallLayouts layoutAdaptor;
    ArrayList layoutslists = new ArrayList();
    SeeallLayouts layoutsAdaptor;
    ImageView back_toolbar_layouts;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_see_all_layout);

        if (AppCompatDelegate.getDefaultNightMode() == AppCompatDelegate.MODE_NIGHT_YES) {
            setTheme(R.style.darkTheme);
            //when dark mode is enabled, we use the dark theme
        } else {
            setTheme(R.style.JobPortaltheam); //default app theme
        }


        if (Build.VERSION.SDK_INT >= 21) {
            Window window = this.getWindow();
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
            window.setStatusBarColor(this.getResources().getColor(R.color.app_blue));
        }

        initilize();

        AsyncTask.execute(new Runnable() {
            @Override
            public void run() {
                fetchads();
                  //fetchlayouts();
            }
        });

    }

    private void initilize() {
        recyclerView = findViewById(R.id.recyclarview_alllayouts);
      //  recyclerView.setHasFixedSize(true);
        GridLayoutManager mgrid = new GridLayoutManager(this,1);
        recyclerView.setLayoutManager(mgrid);
        back_toolbar_layouts = findViewById(R.id.back_toolbar_layouts);

        back_toolbar_layouts.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });
    }

    private void fetchads() {

        DatabaseReference adsimage = FirebaseDatabase.getInstance().getReference().child("layoutsforyou");

        adsimage.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot snapshot) {

                if (snapshot.exists()){

                    HashMap<String, Object> dataMap = (HashMap<String, Object>) snapshot.getValue();
                    for (String key : dataMap.keySet()){
                        Object data = dataMap.get(key);
                        try{

                            HashMap<String, Object> userData = (HashMap<String, Object>) data;

                            layoutlist.add(userData.get(AppConstants.image)+"!!"+userData.get(AppConstants.pid)+"---"+userData.get(AppConstants.description)+"---"+
                                    userData.get(AppConstants.category)+"---"+userData.get(AppConstants.price)+"---"+userData.get(AppConstants.pname)
                                    +"---"+userData.get(AppConstants.propertysize)+"---"+userData.get(AppConstants.location)+"---"+userData.get(AppConstants.number)+"---"+userData.get("type"));

                        }catch (ClassCastException cce){

                            try{
                                String mString = String.valueOf(dataMap.get(key));
                                //addTextToView(mString);
                            }catch (ClassCastException cce2){

                            }
                        }
                    }
                    layoutAdaptor =new SeeallLayouts(layoutlist, SeeAllLayoutActivity.this);
                    RecyclerView.LayoutManager n1layoutManager = new LinearLayoutManager(SeeAllLayoutActivity.this, RecyclerView.VERTICAL, false);
                    recyclerView.setLayoutManager(n1layoutManager);
                    recyclerView.setItemAnimator(new DefaultItemAnimator());

                    mHandler.post(new Runnable() {
                        @Override
                        public void run() {
                            recyclerView.setAdapter(layoutAdaptor);
                        }
                    });
                    layoutAdaptor.notifyItemRangeInserted(0, layoutlist.size());

                }
            }

            @Override
            public void onCancelled(@NonNull DatabaseError error) {

            }
        });
    }

    private void fetchlayouts() {
        DatabaseReference adsimage = FirebaseDatabase.getInstance().getReference().child("layoutsforyou");
        adsimage.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot snapshot) {
                if (snapshot.exists()) {
                    layoutslists.clear();
                    HashMap<String, Object> dataMap = (HashMap<String, Object>) snapshot.getValue();
                    for (String key : dataMap.keySet()) {
                        Object data = dataMap.get(key);
                        try {
                            HashMap userData = (HashMap) data;
                            layoutlist.add(userData.get(AppConstants.image)+"!!"+userData.get(AppConstants.pid)+"---"+userData.get(AppConstants.description)+"---"+
                                    userData.get(AppConstants.category)+"---"+userData.get(AppConstants.price)+"---"+userData.get(AppConstants.pname)
                                    +"---"+userData.get(AppConstants.propertysize)+"---"+userData.get(AppConstants.location)+"---"+userData.get(AppConstants.number)+"---"+userData.get("type"));

                            layoutlist.add(userData.get(AppConstants.image)+"!!"+userData.get(AppConstants.pid)+"---"+userData.get(AppConstants.description)+"---"+
                                    userData.get(AppConstants.category)+"---"+userData.get(AppConstants.price)+"---"+userData.get(AppConstants.pname)
                                    +"---"+userData.get(AppConstants.propertysize)+"---"+userData.get(AppConstants.location)+"---"+userData.get(AppConstants.number));


                        } catch (ClassCastException cce) {

                            try {
                                String mString = String.valueOf(dataMap.get(key));
                                //addTextToView(mString);
                            } catch (ClassCastException cce2) {

                            }
                        }
                    }
                    Collections.shuffle(layoutslists);
                    layoutsAdaptor = new SeeallLayouts(layoutslists, SeeAllLayoutActivity.this);
                    RecyclerView.LayoutManager n1layoutManager = new LinearLayoutManager(SeeAllLayoutActivity.this, RecyclerView.HORIZONTAL, false);
                    recyclerView.setLayoutManager(n1layoutManager);
                    recyclerView.setItemAnimator(new DefaultItemAnimator());

                    mHandler.post(new Runnable() {
                        @Override
                        public void run() {

                          /*  if (progressDialog != null) {
                                if (progressDialog.isShowing()) {
                                    progressDialog.dismiss();
                                }
                            }*/
                            recyclerView.setAdapter(layoutsAdaptor);
                          //  layoutsAdaptor.notifyItemRangeInserted(0, adslist.size());
                        }
                    });
                }
            }

            @Override
            public void onCancelled(@NonNull DatabaseError error) {

            }
        });
    }


    @Override
    public void onClick(View view) {

    }
}

File: app/src/main/java/com/sbd/gbd/Activities/BasicActivitys/SeeAllLayoutActivity.kt
package com.sbd.gbd.Activities.BasicActivitys

import android.os.AsyncTask
import android.os.Build
import android.os.Bundle
import android.os.Handler
import android.view.View
import android.view.WindowManager
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate
import androidx.lifecycle.LifecycleCoroutineScope
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener
import com.sbd.gbd.Adapters.LayoutsAdaptor
import com.sbd.gbd.Adapters.SeeallLayouts
import com.sbd.gbd.Model.LayoutModel
import com.sbd.gbd.R
import com.sbd.gbd.Utilitys.AppConstants
import com.sbd.gbd.Utilitys.FirebaseConnects
import kotlinx.coroutines.launch
import java.util.Collections

class SeeAllLayoutActivity : AppCompatActivity(), View.OnClickListener {
    private var recyclerView: RecyclerView? = null
    var layoutManager: RecyclerView.LayoutManager? = null
    var layoutlist: ArrayList<String> = ArrayList<String>()
    private val mHandler = Handler()
    var layoutAdaptor: SeeallLayouts? = null
    var layoutslists: ArrayList<LayoutModel> = ArrayList<LayoutModel>()
    var layoutsAdaptor: SeeallLayouts? = null
    var back_toolbar_layouts: ImageView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_see_all_layout)
        if (AppCompatDelegate.getDefaultNightMode() == AppCompatDelegate.MODE_NIGHT_YES) {
          //  setTheme(R.style.darkTheme)
            //when dark mode is enabled, we use the dark theme
        } else {
            setTheme(R.style.JobPortaltheam) //default app theme
        }
        if (Build.VERSION.SDK_INT >= 21) {
            val window = this.window
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
            window.statusBarColor = this.resources.getColor(R.color.app_blue)
        }
        initilize()
        lifecycleScope.launch {fetchlayouts()}
    }

    private fun initilize() {
        recyclerView = findViewById(R.id.recyclarview_alllayouts)
        //  recyclerView.setHasFixedSize(true);
        val mgrid = GridLayoutManager(this, 1)
        recyclerView?.setLayoutManager(mgrid)
        back_toolbar_layouts = findViewById(R.id.back_toolbar_layouts)
        back_toolbar_layouts?.setOnClickListener(View.OnClickListener { finish() })
    }

    private fun fetchlayouts() {
        FirebaseConnects.fetchDataWithLayout(AppConstants.layouts,AppConstants.layoutsname) { dataMap ->
            if (dataMap != null) {
                layoutslists.clear()
                for (data in dataMap) {
                    try {
                        val userData = data.value as HashMap<*, *>?
                        if (userData?.get(AppConstants.Status).toString()
                                .equals(AppConstants.user)
                        ){
                            layoutslists.add(
                                LayoutModel(
                                    userData?.get(AppConstants.image).toString(),
                                    userData?.get(AppConstants.image2).toString(),
                                    userData?.get(AppConstants.pid).toString(),
                                    userData?.get(AppConstants.description).toString(),
                                    userData?.get(AppConstants.date).toString(),
                                    userData?.get(AppConstants.type).toString(),
                                    userData?.get(AppConstants.price).toString(),
                                    userData?.get(AppConstants.pname).toString(),
                                    userData?.get(AppConstants.propertysize).toString(),
                                    userData?.get(AppConstants.location).toString(),
                                    userData?.get(AppConstants.number).toString(),
                                    userData?.get(AppConstants.Status).toString(),
                                    userData?.get(AppConstants.sitesAvailable).toString(),
                                    userData?.get(AppConstants.postedBy).toString(),
                                    userData?.get(AppConstants.facing).toString(),
                                    userData?.get(AppConstants.layoutarea).toString(),
                                    userData?.get(AppConstants.point1).toString(),
                                    userData?.get(AppConstants.point2).toString(),
                                    userData?.get(AppConstants.point3).toString(),
                                    userData?.get(AppConstants.point4).toString(),
                                )
                            )
                        }
                    } catch (_: ClassCastException) { }
                }
                layoutAdaptor = SeeallLayouts(layoutslists, this@SeeAllLayoutActivity)
                val n1layoutManager: RecyclerView.LayoutManager =
                    LinearLayoutManager(this@SeeAllLayoutActivity, RecyclerView.VERTICAL, false)
                recyclerView!!.layoutManager = n1layoutManager
                recyclerView!!.itemAnimator = DefaultItemAnimator()
                mHandler.post { recyclerView!!.adapter = layoutAdaptor }
                layoutAdaptor!!.notifyItemRangeInserted(0, layoutlist.size)
            }
        }
    }

    override fun onClick(view: View) {}
}
